cmd/storage/postgresql/external.go

```diff
@@ -0,0 +1,197 @@
+package storage
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	add "github.com/Tanya1515/gophermarket/cmd/additional"
+)
+
+func (db *PostgreSQL) RegisterNewUser(user add.User) error {
+
+	_, err := db.dbConn.Exec("INSERT INTO users (login, password, sum, with_drawn) VALUES($1,crypt($2, gen_salt('xdes')),$3,$4)", user.Login, user.Password, 0, 0)
+
+	if err != nil {
+		return fmt.Errorf("error while inserting user with login %s: %w", user.Login, err)
+	}
+	return nil
+}
+
+func (db *PostgreSQL) AddNewOrder(login string, orderNumber string) (err error) {
+
+	var id string
+
+	rows, err := db.dbConn.Query("SELECT orders.id FROM orders JOIN users ON user_id=users.id WHERE users.login=$1", login)
+	if err != nil {
+		return
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		err = rows.Scan(&id)
+		if err != nil {
+			return
+		}
+		if id == orderNumber {
+			return fmt.Errorf("the order with number %s is in the system", orderNumber)
+		}
+	}
+
+	err = rows.Err()
+	if err != nil {
+		return
+	}
+	row := db.dbConn.QueryRow("SELECT user_id from orders WHERE id=$1", orderNumber)
+	err = row.Scan(&id)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return
+	}
+
+	if err == nil {
+		return fmt.Errorf("error: order with number %s already exists and belongs to another user", orderNumber)
+	}
+
+	_, err = db.dbConn.Exec("INSERT INTO orders (id, status, accrual, UploadedAt, user_id) VALUES($1, $2, $3, $4, (SELECT id FROM users WHERE users.login=$5))", orderNumber, "NEW", 0, time.Now().Format(time.RFC3339), login)
+
+	return
+}
+
+func (db *PostgreSQL) ProcessPayPoints(order add.OrderSpend, login string) (err error) {
+
+	tx, err := db.dbConn.Begin()
+
+	if err != nil {
+		tx.Rollback()
+		return fmt.Errorf("error while starting transaction: %w", err)
+	}
+
+	_, err = tx.Exec("UPDATE users SET sum=sum-$1 WHERE login=$2;", order.Sum, login)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	_, err = tx.Exec("UPDATE users SET with_drawn=with_drawn+$1 WHERE login=$2;", order.Sum, login)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	_, err = tx.Exec("INSERT INTO order_spend (id, ProcessedAt, sum, user_id) VALUES($1, $2, $3, (SELECT id FROM users WHERE login=$4));", order.Number, time.Now().Format(time.RFC3339), order.Sum, login)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+
+	err = tx.Commit()
```

Comment:
```suggestion
	err = tx.Commit()
	if err != nil {
		tx.Rollback()
		return
	}	
```
Ошибка при Commit - повод вызвать Rollback.